openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/remote/ServerConnector.java

```diff
@@ -171,32 +175,40 @@ private void getSecureServerConnection(String user, String pw) {
 		String[] credentials = new String[] { user, pw };
 		props.put("jmx.remote.credentials", credentials);             
 
-		// If we didn't connect keep try - the jvm being monitored might not be 
-		// ready to accept connections yet 
 		boolean connected = false;
 		int attempts = 0;
-		while (connected == false && attempts < 30) {        
+		long connectTimeout = 300L * 1000L * 1000L * 1000L;  // 300 secs in nanoseconds
```

Comment:
Duplicate code block?

Reply:
Please can you be more specific about which "duplicate code block" you mean?

Reply:
```
long connectTimeout = 300L * 1000L * 1000L * 1000L;  // 300 secs in nanoseconds
			try {			long connectStart = System.nanoTime();
				Message.logOut("Trying to connect using JMXConnectorFactory");			long connectEnd = 0L;
		long connectElapsed = 0L;
		while (connected == false && attempts < 30 && (connectElapsed < connectTimeout) ) {
			Message.logOut("Attempting to connect");
			try {    
				connector = JMXConnectorFactory.connect(address);					connector = JMXConnectorFactory.connect(address);
				connected = true;					connected = true;
			} catch (IOException ie) {				} catch (IOException ie) {
				// If we didn't connect, keep trying - the JVM being monitored might not be 					// If we didn't connect keep trying - the jvm being monitored might not be 
				// ready to accept connections yet.					// ready to accept connections yet 
				try {           	  					connectEnd = System.nanoTime();
					Thread.sleep(10000);					connectElapsed = connectEnd - connectStart;
```

In ServerConnector.java line 112 and now here again?

Reply:
Thanks for the clarification.
The class has two similar methods - `getServerConnection()` and `getSecureServerConnection()` with much common code between them.  They could refactored to share much of that code, but that wasn't the focus of this PR.  They both attempt to establish a JMX connection with a max attempts (and now an elapsed timeout) retry capability.  With the code as it is, each method has its own value for attempts and timeout.  They could be the same (as they are now), but different values could in theory be set for the secure vs. non-secure connections.
(There are other ways the JLM tests might benefit from some refactoring too - there are other open issues which advocate that.)